TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
+						self assert: tree right equals: each size ]
+]
+
+{ #category : #'error testing' }
+TPRBParserTest >> testFaultyLiteralRaiseSyntaxError [
+	#('#"bar"foo' '# foo' '#1' '#12' '#12.3' '# 1' '##1' '#"bar"1') 
+		do: [ :expr | 
+		self should: [self parserClass parseExpression: expr] raise: SyntaxErrorNotification ]
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageAssignementOfANonVariable [
+	"Raising a syntax exception when trying to do an assignement on a message."
+	self should: [self parserClass parseExpression: 'receiver selector := 1 + 2'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageAssignementWithMisplacedDot [
+	"The idea is for the parser to realise the first dot isn't supposed to be here and select it. Also add a side note to give suggestion     on the provenance of the error."
+	self should: [self parserClass parseExpression: 'temp := .1 + 2.'] raise: SyntaxErrorNotification.
+	self should: [self parserClass parseExpression: 'temp := 1. + 2.'] raise: SyntaxErrorNotification.
+	self should: [self parserClass parseExpression: 'temp := 1 +. 2.'] raise: SyntaxErrorNotification.
 	
-	self exampleClasses do: [ :class |
-		class selectors do: [ :each |
-			tree := class parseTreeFor: each.
-			self assert: (tree equalTo: tree withMapping: Dictionary new) ] ]
+	"The same could be made for other special characters."
+	self should: [self parserClass parseExpression: 'temp := ;1 + 2.'] raise: SyntaxErrorNotification.
+	self should: [self parserClass parseExpression: 'temp := ,1 + 2.'] raise: SyntaxErrorNotification.
 ]
 
-{ #category : #'tests comparing' }
-TPRBParserTest >> testEquivalentExceptRenaming [
-	#(#('a 3-4' 'a 4-3' false) #('a #[3 4]' 'a #(3 4)' false) #('a variable1 ~~ "comment" variable2' 'a variable1 ~~ variable2' true) #('a variable1' 'a variable2' false) #('a [:a :b | a + b]' 'a [:b :a | a + b]' false) #('a | a b | a + b' 'a | b a | a + b' true) #('a | a | a msg1; msg2' 'a | b | b msg2; msg2' false) #('a c' 'a d' true) #('a | a b | a := b. ^b msg1' 'a | a b | b := a. ^a msg1' true) #('a | a b | a := b. ^b msg1: a' 'a | a b | b := a. ^b msg1: a' false) #('a: b b + 4' 'a: e e + 4' true) #('a: b b + 4' 'b: b b + 4' false) #('a: b b: c b + c' 'a: c b: b c + b' true) #('a: a b: b a + b' 'a: b b: a a + b' false))
-		do: [ :each | 
-			self
-				assert: ((self parserClass parseMethod: each first) equalTo: (self parserClass parseMethod: (each at: 2)) exceptForVariables: #('c'))
-				identicalTo: each last ]
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageBadlyParenthesizedEndOfStatement [
+	"Should propose a closing parenthese before the end of the statement. Should also offer possibility to quickly erase it."
+	self should: [self parserClass parseExpression: '(temp := 1 + 3 .'] raise: SyntaxErrorNotification.
 ]
 
-{ #category : #'tests parsing' }
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageEncouteringEndOfStatement [

```

Comment:
`testInvalidSelector`. Here this piece of code `temp 1 + 3` would be valid if `1` was a unary selector!